### PR TITLE
Create malus_law.py

### PR DESCRIPTION
**Pull request by SparshRastogi**
Finding the intensity of light transmitted through a polariser using Malus Law and by taking initial intensity and angle between polariser and axis as input

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/malus_law.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    if initial_intensity < 0:
+        raise ValueError("The value of intensity cannot be negative")
+        # handling of negative values of initial intensity
+    if angle < 0 or angle > 360:
+        raise ValueError(
+            "In Malus Law,the angle between the polarizer and the axis always lies between 0 degree and 360 degrees"
+        )
+        # handling of values out of allowed range
+    final_intensity = initial_intensity * (math.cos(math.radians(angle)) ** 2)
+    return round(final_intensity, 2)
```
Comment (CaedenPH) [inline physics/malus_law.py:78]: Why round the result?
```suggestion
    return final_intensity
```
You should keep the precision and round it in the tests

File: physics/malus_law.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    >>> malus_law(100,360)
+    100.0
+    """
+
+    if initial_intensity < 0:
+        raise ValueError("The value of intensity cannot be negative")
+        # handling of negative values of initial intensity
+    if angle < 0 or angle > 360:
+        raise ValueError(
+            "In Malus Law,the angle between the polarizer and the axis always lies between 0 degree and 360 degrees"
```
Comment (CaedenPH) [inline physics/malus_law.py:74]:
```suggestion
            "In Malus Law, the angle between the polarizer and the axis always lies between 0 degree and 360 degrees"
```

File: physics/malus_law.py (hunk 1 above)
Comment (CaedenPH) [inline physics/malus_law.py:78]:
```suggestion
    return initial_intensity * (math.cos(math.radians(angle)) ** 2)
```

File: physics/malus_law.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import doctest
```
Comment (CaedenPH) [inline physics/malus_law.py:1]: This is only used in the
`if __name__ == "__main__`
and therefore should be imported there
```suggestion
```

File: physics/malus_law.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    if angle < 0 or angle > 360:
+        raise ValueError(
+            "In Malus Law,the angle between the polarizer and the axis always lies between 0 degree and 360 degrees"
+        )
+        # handling of values out of allowed range
+    final_intensity = initial_intensity * (math.cos(math.radians(angle)) ** 2)
+    return round(final_intensity, 2)
+
+
+if __name__ == "__main__":
```
Comment (CaedenPH) [inline physics/malus_law.py:81]:
```suggestion
if __name__ == "__main__":
    import doctest

```

**Review by CaedenPH — COMMENTED** [review]
@SparshRastogi fix the failing pre-commit (lines too long), commit my changes and then this pr will be ready for merge.
Great work!

File: physics/malus_law.py (hunk 3 above)
Comment (SparshRastogi) [reply physics/malus_law.py:1]: Ok Thanks

File: physics/malus_law.py (hunk 1 above)
Comment (SparshRastogi) [reply physics/malus_law.py]: Yes,thank you for the suggestion,made changes to round in the testcases instead of main code

**Review by CaedenPH — APPROVED** [review]
@cclauss Can we get your review? 
I wasn't completely sure if I should've enforced moving the imports below the docstring, so I left it. Should we enforce docstrings from line 1 and then code/imports after?

**Review by cclauss — APPROVED** [review]

**Pull request merged**